ambry-server/src/test/java/com.github.ambry.server/AmbryRequestsTest.java

```diff
@@ -1053,9 +1156,12 @@ public void shutdown() throws StoreException {
      */
     PartitionId startedPartitionId = null;
 
-    MockStorageManager() throws StoreException {
+    private final Set<StoreKey> validKeysInStore;
```

Comment:
minor: would it be easier/better to store a list of invalid keys?

Reply:
The mock store impl needs to check if a key in `validKeysInStore` or not. Set is faster.

Reply:
Ah set is fine - but I was wondering about what you want to track - valid or invalid keys?

Reply:
Here the assumption is that a key is invalid if it is not explicitly present in `validKeysInStore` - the other approach is that it is invalid only if it is present in `invalidKeysInStore`

Reply:
I feel `validKeysInStore` is more natural. It's like keys are in real store. 
Plus, `invalidKeysInStore ` needs to add key everytime we want to check. Code maybe not friendly.